src/main/java/jenkins/plugins/slack/webhook/CommandRouter.java

```diff
@@ -0,0 +1,125 @@
+package jenkins.plugins.slack.webhook;
+
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import java.lang.reflect.Method;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+
+
+
+
+public class CommandRouter<T> {
+
+    public CommandRouter() { }
+
+    public List<Route> routes = new ArrayList<Route>();
+
+    public CommandRouter addRoute(String regex,
+        String command,
+        String commandDescription,
+        RouterCommand routerCommand) {
+
+        this.routes.add(new CommandRouter.Route(regex,
+            command,
+            commandDescription,
+            routerCommand));
+
+        return this;
+    }
+
+    public List<Route> getRoutes() {
+        return this.routes;
+    }
+
+    public T route(String command) throws CommandRouterException,
+        RouteNotFoundException {
+
+        T message = null;
+
+        for (Route pa : routes) {
+
+            Matcher matcher = pa.regex.matcher(command);
+
+            boolean matches = matcher.matches();
+
+            if (matches) {
+    
+                List<String> parameters = new ArrayList<String>();
+
+                for (int index = 0; index <= matcher.groupCount(); index++) {
+                    parameters.add(matcher.group(index));
+                }
+
+                String[] parametersArray = null;
+
+                //
+                // strip the first matched group
+                //
+                if (parameters.size() > 1) {
+                    parameters.remove(0);
+                    parametersArray = new String[parameters.size()];
+                    parametersArray = parameters.toArray(parametersArray);
+                }
+                
+                try {
+                    message = (T)pa.routerCommand.execute(parametersArray);
```

Comment:
You could just return from here, remove the break out of the loop, and remove the return at the end of the method.